File: Contribute/Plugin Samples/Plugin Samples/IronPython/WavyStylePlugin/SpellChecker.py
from System.IO import *
from System.Drawing import *
from FastColoredTextBoxNS import *
from System.Collections.Generic import *

class SpellChecker(object):
	def __init__(self, fctb, dictionarypath):
		self._wavyStyle = WavyLineStyle(255, Color.Red)
		List = File.ReadAllLines(dictionarypath)
		DictionaryPath = dictionarypath
		self._words = HashSet[str](List, System.StringComparer.InvariantCultureIgnoreCase)
		self._tb = fctb
		fctb = self._tb
		self._tb.TextChangedDelayed += self.tb_TextChangedDelayed

	def get_DictionaryPath(self):
		return self._Dictionary

	def set_DictionaryPath(self, value):
		self._Dictionary = value

	DictionaryPath = property(fget=get_DictionaryPath, fset=set_DictionaryPath)

	def SpellCheck(self, sender):
		self.tb_TextChangedDelayed(sender, TextChangedEventArgs(self._tb.Range))

	def tb_TextChangedDelayed(self, sender, e):
		System.Threading.ThreadPool.QueueUserWorkItem()

	#
	def Clear(self):
		self._tb.Range.ClearStyle(self._wavyStyle)

File: Contribute/Plugin Samples/Plugin Samples/Ruby/WavyStylePlugin/SpellChecker.rb
require "mscorlib"
require "System.IO, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Drawing, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "FastColoredTextBoxNS, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Collections.Generic, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"

module WavyStylePlugin
	class SpellChecker
		def initialize(fctb, dictionarypath)
			@wavyStyle = WavyLineStyle.new(255, Color.Red)
			List = File.ReadAllLines(dictionarypath)
			DictionaryPath = dictionarypath
			@words = HashSet[System::String].new(List, System.StringComparer.InvariantCultureIgnoreCase)
			self.@tb = fctb
			fctb = self.@tb
			self.@tb.TextChangedDelayed { |sender, e| self.tb_TextChangedDelayed(sender, e) }
		end

		def DictionaryPath
			return @Dictionary
		end

		def DictionaryPath=(value)
			@Dictionary = value
		end

		def SpellCheck(sender)
			self.tb_TextChangedDelayed(sender, TextChangedEventArgs.new(self.@tb.Range))
		end

		def tb_TextChangedDelayed(sender, e)
			System.Threading.ThreadPool.QueueUserWorkItem(			list = List[FastColoredTextBoxNS::Range].new()
			enumerator = e.ChangedRange.GetRanges(@"[\w']+").GetEnumerator()
			while enumerator.MoveNext()
				word = enumerator.Current
				if not @words.Contains(word.Text) then
					list.Add(word)
				end
			end
			#
			e.ChangedRange.ClearStyle(@wavyStyle)
			enumerator = list.GetEnumerator()
			while enumerator.MoveNext()
				word = enumerator.Current
				word.SetStyle(@wavyStyle)
			end
)
		end

		def Clear()
			self.@tb.Range.ClearStyle(@wavyStyle)
		end
	end
end